src/cpp/core/LogOptions.cpp

```diff
@@ -69,14 +69,36 @@ namespace log {
 namespace {
 
 // pick default log path location
-#if defined(__APPLE__)
-const FilePath kDefaultLogPath = core::system::xdg::userDataDir().completePath("log");
-#elif defined(RSTUDIO_SERVER)
-const FilePath kDefaultLogPath("/var/log/rstudio/rstudio-server");
+
+FilePath defaultLogPathImpl()
+{
+#ifdef RSTUDIO_SERVER
+   // server: prefer user data directory if we're not running as root
+   if (core::system::effectiveUserIsRoot())
+   {
+      FilePath defaultPath("/var/log/rstudio/rstudio-server");
+      Error error = defaultPath.ensureDirectory();
```

Comment:
Do we need to create this here? It doesn't look like we did before, and I'm not totally confident this will create it with the correct ownership and permissions.

Reply:
That's a good point -- it shouldn't be necessary; however, it does seem like if the default log directory is not writable we can end up in a situation where the log directory is never set, and that isn't reported to the user.

I'll back out this part of the PR.